lib_eio_posix/low_level.ml

```diff
@@ -0,0 +1,200 @@
+open Eio.Std
+
+(* There are some things that should be improved here:
+
+   - Blocking FDs (e.g. stdout) wait for the FD to become ready and then do a blocking operation.
+     This might not succeed, and will block the whole domain in that case.
+     Ideally, all blocking operations should happen in a sys-thread instead.
+
+   - Various other operations, such as listing a directory, should also be done in a sys-thread
+     to avoid high latencies in the main domain. *)
+
+type ty = Read | Write
+
+module Fd = Fd
+
+let await_readable fd =
+  Fd.use_exn "await_readable" fd @@ fun fd ->
+  Sched.enter @@ fun t k ->
+  Sched.await_readable t k fd
+
+let await_writable fd =
+  Fd.use_exn "await_writable" fd @@ fun fd ->
+  Sched.enter @@ fun t k ->
+  Sched.await_writable t k fd
+
+let rec do_nonblocking ty fn fd =
+  try fn fd with
+  | Unix.Unix_error (EINTR, _, _) -> do_nonblocking ty fn fd    (* Just in case *)
+  | Unix.Unix_error((EAGAIN | EWOULDBLOCK | EINPROGRESS), _, _) ->
```

Comment:
hmm I don't think EINPROGRESS should be here, is it just defensive programming or I'm missing something else ?

Reply:
I think you're right - it's only useful for `connect`, which has its own code for this.